src/exec/Byron.hs

```diff
@@ -109,10 +109,11 @@ download tracer genesisBlock epochSlots securityParam db bp = do
     -- Pick a peer from the list of announcers at random and download
     -- the chain.
     let (peer, rndGen') = pickRandom rndGen (btPeers bt)
+        targetHash = CSL.headerHash (btTip bt)
     chain <- atomically $ ChainDB.getCurrentChain db
     traceWith tracer $ mconcat
       [ "Attempting to download chain with hash "
-      , Text.fromString (show tipHash)
+      , Text.fromString (show targetHash)
```

Comment:
Also fixed this misleading log message. It used to be a lie: the current tip in the database was used instead of the peer's tip.